src/kibana/components/index_patterns/_field_formats.js

```diff
@@ -166,9 +167,15 @@ define(function (require) {
       format.convert = _.memoize(format._origConvert);
     }
 
+    function updateDefaultTimezone() {
+      moment.tz.setDefault(config.get('dateFormat:tz'));
+    }
+
     // memoize once config is ready, and every time the date format changes
     $rootScope.$on('init:config', memoizeDateFormat);
     $rootScope.$on('change:config.dateFormat', memoizeDateFormat);
+    $rootScope.$on('init:config', updateDefaultTimezone);
```

Comment:
Is this listener needed? Can you make a single function that handles memoization and setting the initial default